src/net.h

```diff
@@ -421,19 +422,19 @@ struct CombinerAll
     }
 };
 
-// Signals for message handling
-struct CNodeSignals
+// Interface for message handling
+class NetEventsInterface
 {
-    boost::signals2::signal<bool (CNode*, CConnman&, std::atomic<bool>&), CombinerAll> ProcessMessages;
-    boost::signals2::signal<bool (CNode*, CConnman&, std::atomic<bool>&), CombinerAll> SendMessages;
-    boost::signals2::signal<void (CNode*, CConnman&)> InitializeNode;
-    boost::signals2::signal<void (NodeId, bool&)> FinalizeNode;
+protected:
+    NetEventsInterface() = default;
+public:
+    virtual ~NetEventsInterface() = default;
```

Comment:
I must be missing your point. This is just a safety precaution in case anything ever wants to hold an abstract pointer. This may be the case if, for example, we create a vector of message processors in order to run them against each-other.

I can remove it if you'd like, but I'm not understanding the objection to being cautious.